ya-runtime-erigon/src/main.rs

```diff
@@ -246,3 +279,23 @@ fn generate_password_file(
         .stderr(Stdio::null())
         .spawn()
 }
+
+fn parse_json(config: Option<&String>, default_chain_id: &String) -> String {
+    match config {
+        None => default_chain_id.clone(),
+        Some(json) => {
+            let value: Value =
+                serde_json::from_str(json).expect("Cannot parse config, assumes json string");
+            let network = value["network"].as_str().unwrap();
+            let _ = chain_id_valid(network).unwrap();
+            network.to_owned()
+        }
+    }
+}
+
+fn chain_id_valid(chain_id: &str) -> Result<(), String> {
```

Comment:
If we use an enum type for chains/networks we can replace this with `EnumFromStr` macro from [enum_derive](https://docs.rs/enum_derive/0.1.7/enum_derive/) crate.